CMSgov/easi-app
# [EASI-1779] Add system intake document functionality

# EASI-1779

## Changes and Description

- Introduces System Intake Documents

Most of this was copy-paste from TRB Docs, rather than trying to merge the code together (so there may be some duplication that we could remove in the future, but the priority here is unblocking the frontend work!)

## How to test this change

- `scripts/dev up`
- `scripts/dev test:go`
- Ensure no references to the TRB exist (like I said, lots of copy paste!)

## PR Author Review Checklist

  - DB Migrations are created that allow documents to be associated to System Intakes
  - Modifications to our GQL Schema and GQL Resolvers are made that allow documents to be associated to System Intakes
  - These migrations are run in each environment.
    - Not quite, but when this is deployed that'll be the case 😄 


## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
